util/unifyfs-stage/src/unifyfs-stage-transfer.c

```diff
@@ -0,0 +1,376 @@
+		  __func__);
+            return -EINVAL;
+	}
+	// this is using the plan of forming the src and dst strings in
+	// place using and modifying the contents of the line buffer.
+	// If these become independent buffers these lines will change
+	// to sprintf or memcpy or some such.
+	*src = line+1;
+	(ptr_doublequotes[1])[0] = '\0';
+	// current search continues immediately after 2nd
+	// double quote above.
+	CSP = ptr_doublequotes[1]+1;
+	if (CSP[0] != ' ') {
+	    fprintf(stderr, "%s: line contains no deliminting space!\n",
+		    __func__);
+            return -EINVAL;
+	}
+    } else{
+        // the FROM field of this line isn't quoted; it begins
+        // immediately at the beginning of the line and ends
+        // with the first space character
+        *src = line;
+	loc_of_space = strchr(line, ' ');
+	if (loc_of_space == NULL) {
+	    fprintf(stderr, "%s: line contains no deliminting space!\n",
+		  __func__);
+            return -EINVAL;
+	}
+
+	loc_of_space[0] = '\0';
+	CSP = loc_of_space + 1;
+    }
+    // source file has been parsed out and delimited as a string.  Now
+    // starting at CSP after the previous logic, walk forward
+    // to find the destination string.
+    ptr_doublequotes[2] = strchr(CSP+1, '"');
+    if (ptr_doublequotes[2] == NULL) {
+        // there are no remaining doublequote characters on the line
+        // so we just walk to find the beginning of the filename
+        offset = 0;
+	test_char = *(CSP + offset);
+	while (test_char != '\n' &&
+		(test_char == ' ')) {
+	    offset++;
+	}
+	*dst = CSP + offset;
+	loc_of_newline = strchr((*dst), '\n');
+	if (loc_of_newline != NULL) {
+	    loc_of_newline[0] = '\0';
+	}
+	loc_of_space = strchr((*dst), ' ');
+	if (loc_of_space != NULL) {
+	    loc_of_space[0] = '\0';
+	}
+    } else {
+        // if there *is* another doublequote character, the destination
+        // filename begins on the very next character
+	ptr_doublequotes[3] = strchr(ptr_doublequotes[2]+1, '"');
+	if (ptr_doublequotes[1] == NULL) {
+	    fprintf(stderr, "%s: open dst doublequote but no DQ to close!\n",
+		  __func__);
+            return -EINVAL;
+	}
+	*dst = ptr_doublequotes[2] + 1;
+	ptr_doublequotes[3] = '\0';
+    }
+
+    return 0;
+}
+
+int unifyfs_stage_transfer(unifyfs_stage_t* ctx)
+{
+    int ret = 0;
+    int count = 0;
+    FILE* fp = NULL;
+    char* src = NULL;
+    char* dst = NULL;
+    char linebuf[LINE_MAX] = { 0, };
+    struct stat sb = { 0, };
+
+    if (!ctx) {
+        return EINVAL;
+    }
+
+    fp = fopen(ctx->manifest_file, "r");
+    if (!fp) {
+        fprintf(stderr, "failed to open file %s: %s\n",
+                        ctx->manifest_file, strerror(errno));
+        ret = errno;
+        goto out;
+    }
+
+    while (NULL != fgets(linebuf, LINE_MAX-1, fp)) {
+	if (strlen(linebuf) < 5) {
+	    // the manifest file perhaps ends with a couple of characters
+	    // and/or a newline not meant to be a transfer spec.
+            goto out;
+	}
```

Comment:
Allowing stray newlines is ok, but any other stray characters should be flagged as an error.

Reply:
Ok.  I'll re-do this.

Reply:
Fixed in upcoming patch.